tests/test_config.py

```diff
@@ -242,20 +244,36 @@ def test_no_more_dotting():
 
 def test_mode_apply():
 
-    assert configdefaults.filter_mode("DebugMode") == "DebugMode"
+    assert configdefaults._filter_mode("DebugMode") == "DebugMode"
 
     with pytest.raises(ValueError, match="Expected one of"):
-        configdefaults.filter_mode("not_a_mode")
+        configdefaults._filter_mode("not_a_mode")
 
     # test with theano.Mode instance
     import theano.compile.mode
 
     assert (
-        configdefaults.filter_mode(theano.compile.mode.FAST_COMPILE)
+        configdefaults._filter_mode(theano.compile.mode.FAST_COMPILE)
         == theano.compile.mode.FAST_COMPILE
     )
 
 
+def test_config_pickling():
+    # check that the real thing can be pickled
+    root = configdefaults.config
+    pickle.dump(root, io.BytesIO())
```

Comment:
Unit tests that only test for "non-failures" aren't complete tests.  We already have too many of these in the test suite (see #16) and, since they don't actually confirm the expected results, they've allowed numerous bugs to slip through the tests unnoticed.

As a rule of thumb, a test should at least `assert` something about the expected output.  In this case, after `pickle`ing, it should un`pickle` the `pickle`d result and `assert` the presence/value of any properties that required special `pickle`ing steps.

We need to follow this up with a more complete test.  Unfortunately, that will be harder to do after closing this PR.